test/src/main/java/org/axonframework/test/aggregate/ResultValidatorImpl.java

```diff
@@ -66,7 +68,13 @@ public ResultValidator expectEvents(Object... expectedEvents) {
         Iterator<EventMessage<?>> iterator = publishedEvents.iterator();
         for (Object expectedEvent : expectedEvents) {
             EventMessage actualEvent = iterator.next();
-            if (!verifyEventEquality(expectedEvent, actualEvent.getPayload())) {
+            if (EventMessage.class.isInstance(expectedEvent)) {
```

Comment:
I like the addition to be able to check on both payload and metadata, but it kinda feels like we've added additional behaviour to the 'event-payload-matching' function which is not quite clear up front..

Not settled on what would be the best alternative solution, but the first thing that sprung to mind is to have a `expectEventMessages(EventMessage<?>... expectedEventMessages)` function.

Just thinking out loud here, what do you think @bliessens ?

Reply:
From my perspective `ResultValidatorImpl#expectEventsMatching` is very powerful but comes with certain (too much?) complexity: You need a `sequencOf()` messages which are `equalTo()` or adhere to `predicate()` etc ... These assertions are not very trivial to write.

Knowing the aggregate test fixture you already supports interceptors like `CorrelationDataInterceptor` that copy metadata from command to event and the fact aggregates can publish events with `AggregateLifecycle#apply(Object, MetaData)` I would expect it to be much easier to assert expected payload and metadata. 

Overloading the existing method with `ResultValidator#expectEvents(EventMessage<?>... expectedEvents)` would make the added behaviour more visible without changing the semantics.

Adding the behaviour to a new method like `ResultValidator#expectEventMessages(EventMessage<?>...)` makes things too chatty I think.

Reply:
I like your overloading suggestion! 

However, I believe @abuijze was fine with the set up you've got right now. Let's see what he has to say about this.

Reply:
@bliessens would you be open to introduce that overloaded function you've suggested?